lib/commands/sessions.ts

```diff
@@ -20,24 +20,51 @@
  *
  */
 
-export const login = function(user: string, password: string, route: string = '/apps/files') {
+export class User {
+	user: string
+	password: string
+
+	constructor(user: string, password: string = user) {
+		this.user = user
+		this.password = password
+	}
+}
+
+/**
+ * You should always upload files and/or create users
+ * before login, so that the cookies are NOT YET defined.
+ */
+export const login = function(user: User) {
 	cy.clearCookies()
+
+	// Keep sessions active between tests until
+	// we use the new cypress session API
 	Cypress.Cookies.defaults({
 		preserve: /^(oc|nc)/,
 	})
```

Comment:
Could we move this to a separate function?
It might cause unexpected behavior in tests using the session api such as the ones in text

Reply:
Well, session API is broken right now.
Fix has been dispatched a few hours ago on Cypress repo.
https://github.com/cypress-io/cypress/issues/22230

Reply:
https://github.com/cypress-io/cypress/pull/24603

Reply:
Maybe we could still make this optional with a parameter, so with the already present isolation in text we could still wrap the libraries login command into cy.session. But I talked with @max-nextcloud earlier that there might be cases where we also want to break out of the isolation to get some better performance.

Reply:
I see two options imho.
We've been moving forward for a few years with similar but slightly different approaches.
Seeing cypress is fixing this in the upcoming release, shouldn't we just wait and move that to session as soon as they released?

I'm not sure we should spend too much time here, isolation is not how our software is designed I think.
We should not rely on this for testing :)